internal/cmd/base/servers.go

```diff
@@ -606,3 +617,49 @@ func MakeSighupCh() chan struct{} {
 	}()
 	return resultCh
 }
+
+// SetStatusGracePeriodDuration sets the value for
+// StatusGracePeriodDuration.
+//
+// The grace period is the length of time we allow connections to run
+// on a worker in the event of an error sending status updates. The
+// period is defined the length of time since the last successful
+// update.
+//
+// The setting is derived from one of the following, in order:
+//
+//   * Via the supplied value if non-zero.
+//   * BOUNDARY_STATUS_GRACE_PERIOD, if defined, can be set to an
+//   integer value to define the setting.
+//   * If either of these is missing, the default (15 seconds) is
+//   used.
+//
+// The minimum setting for this value is the default setting. Values
+// below this will be reset to the default.
+func (s *Server) SetStatusGracePeriodDuration(value time.Duration) {
+	var result time.Duration
+	switch {
+	case value > 0:
+		result = value
+	case os.Getenv(statusGracePeriodEnvVar) != "":
+		v := os.Getenv(statusGracePeriodEnvVar)
```

Comment:
I'm curious about the need/choice to set this via an envvar.  It's not a typical pattern in Boundary.  Is there some specific use-case that requires envvar support?

Reply:
@jefferai ^^^

Reply:
@jimlambrt The issue here is that really we want this to be a database value so that it's not a per-controller configuration value that might be different on different controllers. So I'm hesitant to add it to config as a stop-gap because then we'd have to remove it later and force users to switch over.

What this really needs is the previously discussed attributes/defaults system -- the same thing that default worker behavior on targets is waiting on, as well as synchronized user preferences. The idea for the moment is it's an env var that is unpublished/undocumented that acts as a relief valve for anyone having issues with the implementation, but that officially we only have a static value right now.

Reply:
@jefferai and I chatted about this... @vancluever can we add a comment that makes it explicit that this is a temporary stopgap measure?  This will make it clear to everyone that the envvar is not a permanent solution and we  will definitely deprecated it in the future in favor of a holistic approach to attributes/defaults system.   ty!

Reply:
Done!